# Copyright (c) 2008, Oracle and/or its affiliates. All rights reserved.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the "Software"),
# to deal in the Software without restriction, including without limitation
# the rights to use, copy, modify, merge, publish, distribute, sublicense,
# and/or sell copies of the Software, and to permit persons to whom the
# Software is furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice (including the next
# paragraph) shall be included in all copies or substantial portions of the
# Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
# DEALINGS IN THE SOFTWARE.
#

Since libXfont is used by programs like mkfontdir & installalias that run
during Live Upgrade from old Solaris releases to new ones, we need to
minimize dependencies on library versions found only on new releases.

This patch uses macros for htons() & ntohs() to avoid creating a dependency
on the Nevada version of libc (SUNW_1.23) which has added those functions to
libc (they were previously in libnsl on older releases).

Since those are already no-op macros on SPARC, this is not needed there.

--- src/fc/fstrans.c	Mon May 12 14:31:00 2008
+++ src/fc/fstrans.c	Fri Sep 12 19:22:31 2008
@@ -19,7 +19,13 @@
  * TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
  * PERFORMANCE OF THIS SOFTWARE.
  */
  
+#ifndef __sparc
+/* inline htons() & ntohs() to avoid dependency on libc version SUNW_1.23 */
+#define htons(x) BSWAP_16(x)
+#define ntohs(x) BSWAP_16(x)
+#endif
+
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
